python/lsst/summit/utils/peekExposure.py

```diff
@@ -0,0 +1,772 @@
+        self.charImage.measurement.slots.apFlux = "base_CircularApertureFlux_70_0"
+        self.charImage.installSimplePsf.fwhm = 5.0
+        self.charImage.installSimplePsf.width = 21
+
+
+class PeekSpecTask(pipeBase.Task):
+    """Spec mode for PeekExposureTask
+
+    For spec mode, we dramatically increase the detection threshold to avoid
+    creating blends with the long spectra objects that appear in these images.
+    We also change the default aperture flux slot to a larger aperture, which
+    helps overcome challenges with lost flux in the interpolated cores of
+    saturated objects.
+    """
+
+    ConfigClass = PeekSpecTaskConfig
+    _DefaultName = "peekSpec"
+
+    def __init__(self, config, **kwargs):
+        super().__init__(config=config, **kwargs)
+        self.makeSubtask("charImage")
+
+    def run(self, exposure):
+        mi = exposure.getMaskedImage()
+        binned = afwMath.binImage(mi, self.config.binSize)
+        exposure.setMaskedImage(binned)
+
+        # Ignore any Psf that comes with exposure.  Force installSimplePsf.
+        exposure.setPsf(None)
+
+        charImageResult = self.charImage.run(exposure)
+
+        return pipeBase.Struct(
+            mode="spec",
+            binSize=self.config.binSize,
+            binnedSourceCat=charImageResult.sourceCat,
+        )
+
+
+class PeekExposureTaskConfig(pexConfig.Config):
+    donutThreshold = pexConfig.Field(
+        dtype=float,
+        default=50.0,
+        doc="Size threshold in pixels for when to switch to donut mode.",
+    )
+    doPhotoFallback = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="If True, fall back to photo mode if spec mode fails.",
+    )
+    donut = pexConfig.ConfigurableField(
+        target=PeekDonutTask,
+        doc="PeekDonut task",
+    )
+    photo = pexConfig.ConfigurableField(
+        target=PeekPhotoTask,
+        doc="PeekPhoto task",
+    )
+    spec = pexConfig.ConfigurableField(
+        target=PeekSpecTask,
+        doc="PeekSpec task",
+    )
+
+
+class PeekExposureTask(pipeBase.Task):
+    """
+    The basic philosophy of PeekExposureTask is to:
+    1) Classify exposures based on metadata into 'donut', 'spec', and 'photo'.
+    2) Run CharacterizeImageTask on the exposure, but with modifications to run
+       as fast as reasonably possible (e.g., turning features off and binning
+       the image), and also some class specific modifications.
+    3) Return a source catalog and some summary statistics.
+    4) Try only to branch in the code based on the metadata, and not on the
+       data itself.  This avoids problematic discontinuities in summary
+       statistics.
+
+    For speed, many items in CharacterizeImageTask have been turned off,
+    including:
+        - PSF measurement (though not moment measurements)
+        - Deblending
+        - Repair
+        - Aperture correction
+        - Background estimation
+        - Noise replacement
+        - Writing to disk
```

Comment:
From this description it sounds like just running detection and measurement manually (I think that's all that's left) would be simpler, especially from a maintenance standpoint, as it will insulate you from upstream changes to a lot of things you don't care about.

Reply:
Sure, I'll look into that.  Using characterizeImage was helpful during development/optimization but you're right that what I've landed on here is much simpler.

Reply:
Okay, refactor worked.  Turns out the full set of subtasks is `installPsf`, `background`, `detection`, `measurement`, and possibly `catalogCalculation` (I didn't check if I actually care about the outputs of that last one).

I think I can make this a little tighter still by integrating the rebinning logic into the trimmed-down CharacterizeImageTask too, but that'll wait until tomorrow.

Reply:
Also, this refactor (which eliminates a whole round of duplicate measurement) + your don't-needlessly-clone-the-exposure suggestion look to make things go from ~1.0 s per ccd to ~0.5 s.  Thanks!

Reply:
:+1:   And I don't think you need `catalogCalculation`; it exists only to run the CModel-Psf "extendedness" star/galaxy classifier.